### ISSUE TITLE
Cannot resolve node.js core modules in Webpack 2

### ISSUE THREAD

**Issue by GProst** [webpack-contrib/worker-loader#55]
After I migrated from webpack 1 it started to throw following errors:

`Module not found: Error: Can't resolve 'path' in /absolute-file-path`

Modules, which cause errors are: `path`, `fs`, `vm`, `crypto` and `constants`

I've figured out that `webpack` can't resolve it only when it required from file, which was processed by `worker-loader`.

Tried to load it via `require(worker-loader!file-name.js)` syntax and via webpack.config.js settings

Here is my webpack.config.js:

```
const config = {
  target: 'electron-renderer',
  context: __dirname,
  entry: { app: './app.js', vendor: [/*vendors*/]},
  cache: true,
  devtool: 'source-map',
  watch: false

  resolve: {
    extensions: ['.js', '.json', '.jsx'],
    modules: ["node_modules"]
  },

  module: {

    rules: [
      {test: /\.html/, loader: 'html-loader'},
      {test: /worker\.js/, loader: 'worker-loader'},
      {
        test: /\.less$/,
        use: ExtractTextPlugin.extract({
          fallback: "style-loader",
          use: ["css-loader?sourceMap", "less-loader?sourceMap"]
        })
      },
      {
        test: /\.css/,
        use: ExtractTextPlugin.extract({
          fallback: "style-loader",
          use: "css-loader?sourceMap"
        })
      },
      {test: /\.(jpg|png|gif|jpeg|svg|otf|ttf|eot|woff)$/, loader: 'url-loader?limit=10000'}
    ]
  },

  plugins: [

    new ExtractTextPlugin('styles.[contenthash].css'),

    new webpack.optimize.CommonsChunkPlugin({
      names: ['commons', 'vendor', 'manifest'],
      minChuncks: Infinity
    }),

    new HtmlWebpackPlugin({
      template: './index.html',
      filename: 'index.html',
      hash: false,
      inject: 'head',
      cashe: true,
      showErrors: true
    })

  ],

  output: {
    publicPath: './',
    path: path.join(__dirname, 'dist'),
    filename: '[name].[chunkhash].js',
    chunkFilename: '[name].[chunkhash].js'
  }
};

module.exports = config;
```

**Comment by GProst** [comment]
I thought it could be because of [`nvm`](https://github.com/creationix/nvm) which I use, but I switched to system nodejs via `nvm use system` command and webpack still throws these errors.

I also thought it could be `target` property, so I changed it to `node`, but it didn't help too.

**Comment by GProst** [comment]
It looks like it referenced to #44

**Comment by TrySound** [comment]
@GProst Does it still work with webpack 1

**Comment by GProst** [comment]
@TrySound it worked when I was on webpack 1, if that's what you asked

**Comment by TrySound** [comment]
I mean current version of worker loader with webpack 1.

**Comment by GProst** [comment]
@TrySound yep, I've just checked it. 0.8.0 version seems to work fine with webpack 1. Btw, updated up all of my loaders when migrated.

**Comment by mdh-cc** [comment]
Hi,

I ran into the same problem today.  Any progress on this one so far?

**Comment by TrySound** [comment]
@mdh-cc Not yet. I will be glad to accept a PR.

**Comment by Zuzon** [comment]
any news?

**Comment by evilebottnawi** [comment]
@Zuzon can your create minimal reproducible test repo?

**Comment by Zuzon** [comment]
in my case:
webpack build target is 'node-webkit' and when i try to import 'path' module in worker

**Comment by mikadev001** [comment]
+1
(using electron)

main.js
```js
const Worker = require('example.worker.js')
const workerInstance = new Worker()
```

example.worker.js
```js
import fs from 'fs'
console.log(fs)
```

webpack config
```js
var path = require('path')
var webpack = require('webpack')

module.exports = {
  output: {
    path: path.resolve(__dirname, './dist'),
    filename: 'build.js'
  },
  entry: {
    bundle: './app/main.js'
  },
  module: {
      loaders: [
          {
              test: /\.vue$/,
              loader: 'vue-loader'
          },
          {
              test: /\.js$/,
              loader: 'babel-loader',
              exclude: /node_modules/
          },
          {
              test: /\.css$/,
              loader: 'style-loader!css-loader'
          },
          {
              test: /\.worker\.js$/,
              loader: 'worker-loader?inline'
          }
      ]
  },
  plugins: [
      new webpack.ExternalsPlugin('commonjs', [
          'electron', 'vue-router'
      ])
  ],
  resolve: {
    alias: {
      vue: 'vue/dist/vue.js',
      'vue$': 'vue/dist/vue.esm.js' // 'vue/dist/vue.common.js' for webpack 1
    }
  },
  node: {
    fs: 'empty',
    path: 'empty',
    __dirname: true,
    __filename: true
  },
  target: "electron"
}
```
 

Webpack output:
```
ERROR in ./example.worker.js
    Module not found: Error: Can't resolve 'fs' in 'G:\app\testApp\'
     @ ./example.worker.js 1:0-19
```

**Comment by Zuzon** [comment]
it's possible with nwjs since 0.18.4

**Comment by FrogTheFrog** [comment]
Any news regarding this issue?

**Comment by ahutchings** [comment]
Can this be closed since #76 was merged and included in 1.0.0?
